.github/workflows/auto-assign.yml

```diff
@@ -0,0 +1,59 @@
+name: Auto assign pull requests
+
+on:
+  pull_request:
+    types:
+      - opened
+
+jobs:
+  update_pr:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: tibdex/github-app-token@v1.6.0
+        id: generate-token
+        with:
+          app_id: ${{ secrets.METABASE_BOT_APP_ID }}
+          private_key: ${{ secrets.METABASE_BOT_APP_PRIVATE_KEY }}
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Assign author and their team. Returns true if team exists.
+        id: auto-assign
+        uses: actions/github-script@v6
+        with:
+          github-token: ${{ steps.generate-token.outputs.token }}
+          script: |
+            const fs = require('fs');
+            const teamConfig = JSON.parse(fs.readFileSync('.github/team.json', 'utf-8'));
+            const prAuthor = context.payload.pull_request.user.login;
+            const prNumber = context.payload.pull_request.number;
+
+            // Assign PR author
+            await github.rest.issues.addAssignees({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              issue_number: prNumber,
+              assignees: [prAuthor]
+            });
+
+            const team = teamConfig.teams.find(t => t.members.includes(prAuthor));
+            if (!team) {
+              console.log('You are not assigned to any team. If you need one, update .github/team.json');
```

Comment:
nit: it worth having a comment posted on the PR

Reply:
We have a vast portion of PRs that don't belong to any eng team so this would be bothering